sentry-sidekiq/lib/sentry/sidekiq-scheduler/scheduler.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+return unless defined?(::SidekiqScheduler::Scheduler)
+
+module Sentry
+  module SidekiqScheduler
+    module Scheduler
+      def new_job(name, interval_type, config, schedule, options)
+        # Schedule the job upstream first
+        # SidekiqScheduler does not validate schedules
+        # It will fail with an error if the schedule in the config is invalid.
+        # If this errors out, let it fall through.
+        rufus_job = super
+
+        # Constantize the job class, and fail gracefully if it could not be found
+        klass_const =
+        begin
+          config.fetch("class").constantize
+        rescue NameError
+          return rufus_job
+        end
+
+        # For cron, every, or interval jobs — grab their schedule.
+        # Rufus::Scheduler::EveryJob stores it's frequency in seconds, 
+        # so we convert it to minutes before passing in to the monitor.
+        monitor_config = case interval_type
+          when "cron"
+            Sentry::Cron::MonitorConfig.from_crontab(schedule)
+          when "every", "interval" 
+            Sentry::Cron::MonitorConfig.from_interval(rufus_job.frequency.to_f / 60.0, :minute)
+        end
+
+        # If we couldn't build a monitor config, it's either an error, or
+        # it's a one-time job (interval_type is in, or at), in which case
+        # we should not make a monitof for it automaticaly.
+        return rufus_job if monitor_config.nil?
+
+        # only patch if not explicitly included in job by user
+        unless klass_const.send(:ancestors).include?(Sentry::Cron::MonitorCheckIns)
```

Comment:
Is it possible to patch `load_schedules!` to patch all jobs at once instead of going through these conditions repeatedly?

Reply:
this check is still necessary if some user has included the mixin and overridden the slug themselves

Reply:
Actually, I can see that `new_job` is called for new rufus jobs instead of the sidekiq jobs, so it won't be called as often as I thought. Please ignore this.